compiler/rustc_parse/src/lib.rs

```diff
@@ -172,20 +162,26 @@ fn new_parser_from_source_file(
     Ok(parser)
 }
 
+/// Given a source string, produces a sequence of token trees.
+///
+/// NOTE: This only strips shebangs, not frontmatter!
 pub fn source_str_to_stream(
     psess: &ParseSess,
     name: FileName,
     source: String,
     override_span: Option<Span>,
```

Comment:
I might add `strip_tokens: StripTokens` as a parameter to this function in a future PR. At the time of writing, it wouldn't really matter but there are same call sites like the one for `-Zcrate-attr` where we could pass `StripTokens::Nothing` (since the input is wrapped in `#![` `]` anyway) but it wouldn't lead to any observable difference in rustc's behavior.